File: src/main/java/hiiragi283/api/recipe/HiiragiRecipe.java
package hiiragi283.api.recipe;

import mezz.jei.api.recipe.IRecipeWrapper;
import net.minecraftforge.fml.common.Optional;
import net.minecraftforge.registries.IForgeRegistryEntry;

@Optional.Interface(iface = "mezz.jei.api.recipe.IRecipeWrapper", modid = "jei")
public abstract class HiiragiRecipe<T extends IForgeRegistryEntry<T>> extends IForgeRegistryEntry.Impl<T> implements IRecipeWrapper {
}

File: src/main/kotlin/hiiragi283/api/recipe/HiiragiRecipe.kt
package hiiragi283.api.recipe

import mezz.jei.api.recipe.IRecipeWrapper
import net.minecraftforge.fml.common.Optional
import net.minecraftforge.registries.IForgeRegistryEntry

@Optional.Interface(iface = "mezz.jei.api.recipe.IRecipeWrapper", modid = "jei")
abstract class HiiragiRecipe<T : IForgeRegistryEntry<T>?> : IForgeRegistryEntry.Impl<T>(), IRecipeWrapper 